### PR TITLE
MK-550 index attachments

### PR DESCRIPTION
**Pull request by apruden**

### CODE REVIEW

File: mica-core/src/main/java/org/obiba/mica/file/service/FileSystemService.java [hunk 1]
```diff
@@ -364,6 +370,13 @@ public void studyUnpublished(StudyUnpublishedEvent event) {
     publish(String.format("/study/%s", event.getPersistable().getId()), false);
   }
 
+  @Async
+  @Subscribe
+  public void draftStufyUpdated(DraftStudyUpdatedEvent event) {
+    log.info("Draft study {} was update", event.getPersistable());
+    publish(String.format("/study/%s", event.getPersistable().getId()), false);
```
Comment (ymarcon) [inline mica-core/src/main/java/org/obiba/mica/file/service/FileSystemService.java]: Why publishing on update?

File: mica-search/src/main/java/org/obiba/mica/file/search/rest/AbstractFileSearchResource.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  @Path("/{path:.*}")
+  public List<Mica.FileDto> searchFile(@PathParam("path") String path, @QueryParam("query") String query,
+    @QueryParam("recursively") @DefaultValue("false") boolean recursively, @QueryParam("from") @DefaultValue("0") int from,
+    @QueryParam("limit") @DefaultValue(MAX_SIZE) int limit, @QueryParam("sort") @DefaultValue(DEFAULT_SORT) String sort,
+    @QueryParam("order") @DefaultValue("desc") String order) {
+    String queryString = getQueryString(path, query, recursively);
+    PublishedDocumentService.Documents<AttachmentState> states = getDocumentService()
+      .find(from, limit, sort, order, null, queryString);
+
+    return states.getList().stream().map(state -> dtos.asFileDto(state, state.isPublished()))
```
Comment (ymarcon) [inline mica-search/src/main/java/org/obiba/mica/file/search/rest/AbstractFileSearchResource.java]: asFileDto() 2nd param is not the publication status of the AttachmentState, it is the context of the execution: draft file system or published file system.
In the draft file system you return more info in the dto. In the published file system you return less info and results must be published files.

File: mica-search/src/main/java/org/obiba/mica/file/search/FileIndexer.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  @Inject
+  private ElasticSearchIndexer elasticSearchIndexer;
+
+  @Async
+  @Subscribe
+  public void onFilePublished(FilePublishedEvent event) {
+    log.info("File {} was published", event.getPersistable());
+    elasticSearchIndexer.delete(ATTACHMENT_DRAFT_INDEX, event.getPersistable());
+    elasticSearchIndexer.index(ATTACHMENT_PUBLISHED_INDEX, event.getPersistable());
```
Comment (ymarcon) [inline mica-search/src/main/java/org/obiba/mica/file/search/FileIndexer.java]: published file exists in the two file index: draft and published.

**Comment by apruden** [comment]
PR updated

**Pull request merged**